File: judge/java/tests/coin_change.java
package tests;
import java.util.*;
import java.lang.*;
import java.io.*;
import judge.*;
import datastruct.*;

public class coin_change {
    public static int num_test = 5;
    public static int[][] in_0;
    public static int[][] in_org_0;
    public static int[] in_1;
    public static int[] in_org_1;
    public static int[] out;


    public static void load_test() {
        File fil = new File("judge/tests/coin-change.txt");
        FileReader inputFil = null;
        try {
            inputFil = new FileReader(fil);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        BufferedReader in = new BufferedReader(inputFil);
        try {
            in_0 = common.read_int_matrix(in);
            in_org_0 = test_common.copy(in_0);
            in_1 = common.read_int_array(in);
            in_org_1 = test_common.copy(in_1);
            out = common.read_int_array(in);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void judge() {
        load_test();
        common.capture_stdout();
        Solution s = new Solution();

        long startTime = System.currentTimeMillis();

        for(int i = 0; i < num_test; ++i) {
            System.out.printf("Testing case #%d\n", i+1);
            int answer = s.min_coins(in_0[i], in_1[i]);
            if(answer != out[i]) {
                common.release_stdout();
                System.out.printf("%d / %d;", i+1, num_test);
                String outs = common.to_string(coin_change.in_org_0[i]) + ", " + common.to_string(coin_change.in_org_1[i]);
                System.out.print(outs + ";");
                System.out.print(common.to_string(answer) + ";");
                System.out.println(common.to_string(out[i]));
                return;
            }
        }

        common.release_stdout();
        long estimatedTime = System.currentTimeMillis() - startTime;
        System.out.print("Accepted;");
        System.out.println(estimatedTime);
    }
}

File: judge/scala/tests/coin_change.scala
package test
import scala.io.Source
import judge.common
import judge.Solution

object coin_change {
    val num_test = 302;
    var in_0 = List[List[Int]]();
    var in_1 = List[Int]();
    var out = List[]();


    def load_test() = {
        val in = Source.fromFile("./judge/tests/coin-change.txt").getLines;
        in_0 = common.read_int_matrix(in);
        in_1 = common.read_int_array(in);
        out = common.read_int_array(in);
    }

    def judge(): Int = {
        load_test();
        common.capture_stdout();

        val startTime = System.currentTimeMillis();
        var i = 0;

        while(i < num_test) {
            printf("Testing case #%d\n", i+1);
            val answer = Solution.min_coins(in_0(i), in_1(i));
            if() {
                common.release_stdout();
                printf("%d / %d;", i+1, num_test);
                var outs = common.to_string(coin_change.in_0(i)) + ", " + common.to_string(coin_change.in_1(i));
                print(outs + ";");
                print(common.to_string(answer) + ";");
                println(common.to_string(out(i)));
                return 1;
            }
            i += 1;
        }

        common.release_stdout();
        val estimatedTime = System.currentTimeMillis - startTime;
        print("Accepted;");
        println(estimatedTime);
        return 0;
    }
}
